doc/plugin-criteria.md

```diff
@@ -0,0 +1,51 @@
+Krew plugin criteria
+===
+
+Rationale
+---
+
+Krew is a `kubectl` plugin manager.
+
+A plugin provides a technical solution to at least one _task_ concerning a problem around kubernetes.
```

Comment:
The first paragraph (above) is talking about Krew. This paragraph jumps to talking about "A plugin", by which I take it to mean "A kubectl plugin". The description is then very opinionated.

I think it's fine to have strong opinions about what goes in the Krew index. I don't think it's OK to try and generalise that to the open plugin system that kubectl has. Anyone can create any plugin for kubectl that they want, simply by giving it a specific name and putting it on the path. Because of that the Krew index will always be a subset of available Kubectl plugins, which I think is fine.

I'm not sure if you want to define a "Krew plugin", or whether this should read something like "A plugin +in the Krew index+" or similar.

Reply:
Yes, I fully agree. We should make clear that this is about krew plugins only. Can't do it right now, but you're welcome to suggest better wording.

Reply:
I recommend the following wording:

> The centralized Krew plugin index (`krew-index` repository) holds plugins that are applicable to a broad set of kubectl users.
>
> A plugin in Krew plugin index must add new functionality or enhance the existing functionality in kubectl, or capsulate a common operator/developer workflow for higher productivity.

We don't want to exclude people being creative and trying interesting things.

Reply:
Yeah, I like the approach to argue from the user-basis as opposed to `kubectl`-centric.